### ISSUE TITLE
saltutil.runner jobs.list_jobs is broken

### ISSUE THREAD

**Issue by jfindlay** [saltstack/salt#21650]
As described here:
```
salt '*' sys.doc saltutil.runner
'saltutil.runner:'

    Execute a runner module (this function must be run on the master)

    New in version 2014.7

    name
        The name of the function to run
    kwargs
        Any keyword arguments to pass to the runner function

    CLI Example:

        salt '*' saltutil.runner jobs.list_jobs
```

### versions
```yaml
centos-7-main salt (325723e...) # lsb_release -a
LSB Version:    :core-4.1-amd64:core-4.1-noarch:cxx-4.1-amd64:cxx-4.1-noarch:desktop-4.1-amd64:desktop-4.1-noarch:languages-4.1-amd64:languages-4.1-noarch:printing-4.1-amd64:printing-4.1-noarch
Distributor ID: CentOS
Description:    CentOS Linux release 7.0.1406 (Core)
Release:        7.0.1406
Codename:       Core
```
```yaml
centos-7-main salt (325723e...) # salt --versions
           Salt: 2015.2.0rc1-884-g325723e
         Python: 2.7.5 (default, Jun 17 2014, 18:11:42)
         Jinja2: 2.7.3
       M2Crypto: 0.21.1
 msgpack-python: 0.4.4
   msgpack-pure: Not Installed
       pycrypto: 2.6.1
        libnacl: Not Installed
         PyYAML: 3.11
          ioflo: Not Installed
          PyZMQ: 14.3.1
           RAET: Not Installed
            ZMQ: 3.2.5
           Mako: Not Installed
```

### execution
```
centos-7-main salt (325723e...) # salt '*' saltutil.runner jobs.list_jobs
centos-7-main:
    ERROR executing 'saltutil.runner': The following keyword arguments are not valid: __pub_user=root, __pub_arg=['jobs.list_jobs'], __pub_fun=saltutil.runner, __pub_jid=20150313152931163571, __pub_tgt=*, __pub_tgt_type=glob, __pub_ret=
```

### master log
```python
[INFO    ] Clear payload received with command publish
[DEBUG   ] Sending event - data = {'_stamp': '2015-03-13T21:29:31.164058', 'minions': ['ubuntu-14', 'centos-7-main']}
[DEBUG   ] Sending event - data = {'tgt_type': 'glob', 'jid': '20150313152931163571', 'tgt': '*', '_stamp': '2015-03-13T21:29:31.164416', 'user': 'root', 'arg': ['jobs.list_jobs'], 'fun': 'saltutil.runner', 'minions': ['ubuntu-14', 'centos-7-main']}
[INFO    ] User root Published command saltutil.runner with jid 20150313152931163571
[DEBUG   ] Published command details {'tgt_type': 'glob', 'jid': '20150313152931163571', 'tgt': '*', 'ret': '', 'user': 'root', 'arg': ['jobs.list_jobs'], 'fun': 'saltutil.runner'}
[INFO    ] AES payload received with command _return
[INFO    ] Got return from ubuntu-14 for job 20150313152931163571
[DEBUG   ] Sending event - data = {'fun_args': ['jobs.list_jobs'], 'jid': '20150313152931163571', 'return': "ERROR executing 'saltutil.runner': The following keyword arguments are not valid: __pub_user=root, __pub_arg=['jobs.list_jobs'], __pub_fun=saltutil.runner, __pub_jid=20150313152931163571, __pub_tgt=*, __pub_tgt_type=glob, __pub_ret=", 'success': False, 'cmd': '_return', '_stamp': '2015-03-13T21:29:31.187809', 'fun': 'saltutil.runner', 'id': 'ubuntu-14', 'out': 'nested'}
[INFO    ] AES payload received with command _return
[INFO    ] Got return from centos-7-main for job 20150313152931163571
[DEBUG   ] Sending event - data = {'fun_args': ['jobs.list_jobs'], 'jid': '20150313152931163571', 'return': "ERROR executing 'saltutil.runner': The following keyword arguments are not valid: __pub_user=root, __pub_arg=['jobs.list_jobs'], __pub_fun=saltutil.runner, __pub_jid=20150313152931163571, __pub_tgt=*, __pub_tgt_type=glob, __pub_ret=", 'success': False, 'cmd': '_return', '_stamp': '2015-03-13T21:29:31.251154', 'fun': 'saltutil.runner', 'id': 'centos-7-main', 'out': 'nested'}
[DEBUG   ] Reading configuration from /etc/salt/master
[DEBUG   ] Including configuration from '/etc/salt/minion.d/_schedule.conf'
[DEBUG   ] Reading configuration from /etc/salt/minion.d/_schedule.conf
[DEBUG   ] Using cached minion ID from /etc/salt/minion_id: centos-7-main
[DEBUG   ] Reading configuration from /etc/salt/master
[DEBUG   ] LazyLoaded local_cache.clean_old_jobs
[DEBUG   ] Could not LazyLoad config.merge
[DEBUG   ] Updating roots fileserver cache
[DEBUG   ] This salt-master instance has accepted 2 minion keys.
```

### minion log
```python
[INFO    ] User root Executing command saltutil.runner with jid 20150313153034986113
[DEBUG   ] Command details {'tgt_type': 'glob', 'jid': '20150313153034986113', 'tgt': '*', 'ret': '', 'user': 'root', 'arg': ['jobs.list_jobs'], 'fun': 'saltutil.runner'}
[INFO    ] Starting a new job with PID 23161
[DEBUG   ] LazyLoaded saltutil.runner
[DEBUG   ] LazyLoaded jobs.list_jobs
[ERROR   ] Problem executing 'saltutil.runner': The following keyword arguments are not valid: __pub_user=root, __pub_arg=['jobs.list_jobs'], __pub_fun=saltutil.runner, __pub_jid=20150313153034986113, __pub_tgt=*, __pub_tgt_type=glob, __pub_ret=
Traceback (most recent call last):
  File "/root/salt/salt/minion.py", line 1112, in _thread_return
    return_data = func(*args, **kwargs)
  File "/root/salt/salt/modules/saltutil.py", line 769, in runner
    return rclient.cmd(fun, [], kwarg=kwargs)
  File "/root/salt/salt/client/mixins.py", line 206, in cmd
    self.functions[fun], arglist, pub_data
  File "/root/salt/salt/minion.py", line 278, in load_args_and_kwargs
    .format(', '.join(invalid_kwargs))
SaltInvocationError: The following keyword arguments are not valid: __pub_user=root, __pub_arg=['jobs.list_jobs'], __pub_fun=saltutil.runner, __pub_jid=20150313153034986113, __pub_tgt=*, __pub_tgt_type=glob, __pub_ret=
[INFO    ] Returning information for job: 20150313153034986113
[DEBUG   ] Re-using SAuth for ('/etc/salt/pki/minion', 'centos-7-main', 'tcp://127.0.0.1:4506')
```

**Comment by rallytime** [comment]
@jfindlay Can you give this another try. I think this has been fixed at the head of 2015.2, says @thatch45.

**Comment by jfindlay** [comment]
I've tested again and get the same error for both 2014.7 and 2015.2.

**Comment by basepi** [comment]
@jfindlay Can you try cleaning your cache and try again? @UtahDave and I spent a couple hours chasing a completely different list_jobs issue and it was working great for us in 2015.2. Neither of us were getting this error, so I'm wondering if it may be stale files or something.

**Comment by basepi** [comment]
Ohhhhh, this is saltutil.runner....  /me re-reads.

**Comment by basepi** [comment]
Fixed in #21849. I'll make sure it gets merged forward today too.
